bluebutton-server-app/src/main/java/gov/hhs/cms/bluebutton/server/app/stu3/providers/TransformerUtils.java

```
@@ -2491,4 +2519,75 @@ static void mapHcpcs(ExplanationOfBenefit eob, ItemComponent item, Optional<Char
+	    
+		if (ndcProductMap.containsKey(claimDrugCodeReformatted)) {
+			String ndcSubstanceName = ndcProductMap.get(claimDrugCodeReformatted);
+			return ndcSubstanceName;
+		}
+
+		if (!drugCodeLookupMissingFailures.contains(claimDrugCodeReformatted)) {
+			drugCodeLookupMissingFailures.add(claimDrugCodeReformatted);
+			LOGGER.info("No national drug code value (PRODUCTNDC column) match found for drug code {} in resource {}.",
+					claimDrugCodeReformatted, "fda_products_utf8.tsv");
+		}
+
+		return null;
+	}
+
+	/**
+	 * Reads ALL the PRODUCTNDC and SUBSTANCENAME fields from the FDA NDC Products
+	 * file which was downloaded during the build process
+	 * 
+	 */
+	public static Map<String, String> readFDADrugCodeFile() {
+
+		InputStream ndcProductStream = Thread.currentThread().getContextClassLoader()
+				.getResourceAsStream("fda_products_utf8.tsv");
+
+		BufferedReader ndcProductsIn = null;
+		ndcProductsIn = new BufferedReader(new InputStreamReader(ndcProductStream));
+
+		// We want to extract the PRODUCTNDC and SUBSTANCENAME from the FDA Products
+		// file (fda_products_utf8.tsv is in /target/classes directory) and put in a Map
+		// for easy retrieval to get the
+		// display value which is the SUBSTANCENAME
+		String line = "";
+		try {
+			ndcProductsIn.readLine();
+			while ((line = ndcProductsIn.readLine()) != null) {
+				String ndcProductColumns[] = line.split("\t");
+				String nationalDrugCodeManufacturer = StringUtils
+						.leftPad(ndcProductColumns[1].substring(0, ndcProductColumns[1].indexOf("-")), 5, '0');
+				String nationalDrugCodeIngredient = StringUtils.leftPad(ndcProductColumns[1]
+						.substring(ndcProductColumns[1].indexOf("-") + 1, ndcProductColumns[1].length()), 4, '0');
+				ndcProductMap.put(nationalDrugCodeManufacturer + "-" + nationalDrugCodeIngredient,
+						ndcProductColumns[13]);
+			}
+			ndcProductsIn.close();
+		} catch (IOException e) {
+			e.printStackTrace();
```

This is definitely a bug: you should never swallow/ignore exceptions like this. Always go boom.